File: 2551-apply-operations-to-an-array/2551-apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        newNums = [0] * len(nums)
        count = 0
        i = 0

        while i < len(nums) - 1:
            if nums[i] != 0:
                if nums[i] == nums[i + 1]:
                    newNums[count] = nums[i] * 2
                    i += 1
                else:
                    newNums[count] = nums[i]
                
                count += 1
            i += 1
        
        if i < len(nums) and nums[i] != 0:
            newNums[count] = nums[i]

        return newNums

File: 2551-apply-operations-to-an-array/2551-apply-operations-to-an-array.ts
function applyOperations(nums: number[]): number[] {
    const r = nums.length - 1;
    for (let i = 0; i < r; i++) {
        if (nums[i] === nums[i + 1]) {
            nums[i] *= 2;
            nums[i + 1] = 0;
        }
    }
    const zeros = nums.filter(v => v === 0).length;
    nums = nums.filter(v => v > 0)

    for (let j = 0; j < zeros; j++) {
        nums.push(0)
    }
    return nums
};